### ISSUE TITLE
Crash on rapid join/leave sequency

### ISSUE THREAD

**Issue by maxaudron** [r-lmr/lmrdashboard#64]
Crash log:
```log
Hayate-kun has come online.
[ RowDataPacket { user: 'Hayate-kun' } ] RowDataPacket { user: 'Hayate-kun' }
Hayate-kun has come online.
User Hayate-kun has parted.
[] undefined

/lmrdashboard/backend/irc/utils/db/Users.ts:19
        await knex('online_users').del().where({ user: retrievedUser[0]['user'] });
                                                                       ^
TypeError: Cannot read property 'user' of undefined
    at Function.<anonymous> (/lmrdashboard/backend/irc/utils/db/Users.ts:19:72)
    at step (/lmrdashboard/backend/irc/utils/db/Users.ts:33:23)
    at Object.next (/lmrdashboard/backend/irc/utils/db/Users.ts:14:53)
    at /lmrdashboard/backend/irc/utils/db/Users.ts:8:71
    at new Promise (<anonymous>)
    at __awaiter (/lmrdashboard/backend/irc/utils/db/Users.ts:4:12)
    at /lmrdashboard/backend/irc/utils/db/Users.ts:17:33
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

what happened in IRC:
```
[12:07:41 pm] --> Hayate-kun (kvirc@irc-k304n9.sgdh.0vn3.28v05a.IP) has joined #linuxmasterrace
[12:07:42 pm] <-- Hayate-kun (kvirc@irc-k304n9.sgdh.0vn3.28v05a.IP) has quit (Changing host)
[12:07:42 pm] --> Hayate-kun (kvirc@snoonet.org/user/Hayate-kun) has joined #linuxmasterrace
[12:26:44 pm] <-- Hayate-kun (kvirc@snoonet.org/user/Hayate-kun) has quit (Quit: KVIrc 5.0.0 Aria http://www.kvirc.net/)
```

**Comment by aboft** [comment]
@maxaudron @LoLei this seems to be an issue where it parsed 2 joins first and then tried deleting twice instead of join -> delete -> join -> delete. It is also possible that the IRC server sent commands of JOIN -> JOIN -> QUIT -> QUIT due to the IRC logs. We could maybe enable some debugging logs for a bit to see how the IRC is transmitting and we can further verify.

The below code should prevent the connection from crashing.

```
static async deleteUser(nick: string, server: string) {
    await knex('online_users')
      .select('user')
      .where('user', 'like', `%${nick}`)
      .andWhere({ server })
      .orderBy('user', 'asc')
      .then(async (retrievedUser) => {
        if (retrievedUser.length > 1 && retrievedUser[0]) {
          console.log(retrievedUser, retrievedUser[0]);
          await knex('online_users').del().where({ user: retrievedUser[0]['user'] });
        }
      })
      .catch((e) => console.log('ERROR IN Users.deleteUser', e));
    console.log(`User ${nick} has parted.`);
  }
```

**Comment by maxaudron** [comment]
```
Saving message to db.
R0flcopt3r AA why isn't anything getting displayed?!
Saving message to db.
[] undefined

/lmrdashboard/backend/irc/utils/db/Users.ts:19
        await knex('online_users').del().where({ user: retrievedUser[0]['user'] });
                                                                       ^
TypeError: Cannot read property 'user' of undefined
    at Function.<anonymous> (/lmrdashboard/backend/irc/utils/db/Users.ts:19:72)
    at step (/lmrdashboard/backend/irc/utils/db/Users.ts:33:23)
    at Object.next (/lmrdashboard/backend/irc/utils/db/Users.ts:14:53)
    at /lmrdashboard/backend/irc/utils/db/Users.ts:8:71
    at new Promise (<anonymous>)
    at __awaiter (/lmrdashboard/backend/irc/utils/db/Users.ts:4:12)
    at /lmrdashboard/backend/irc/utils/db/Users.ts:17:33
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by aboft** [comment]
fixed with #69
